Plutarch/Internal.hs

```diff
@@ -82,64 +96,61 @@ data PDelayed (a :: k -> Type) (s :: k)
 
 plam' :: (Term s a -> Term s b) -> Term s (a :--> b)
 plam' f = Term $ \i ->
-  let v = Term $ \j -> (RVar (j - (i + 1)), [])
-      (t, deps) = asRawTerm (f v) (i + 1)
-   in (RLamAbs $ t, deps)
+  let v = Term $ \j -> TermResult (RVar (j - (i + 1))) []
```

Comment:
```hs
TermResult (RVar (j - (i + 1))) []
```
Is there a reason for not using `mkNew` here?

Reply:
Nice catch.